### PR TITLE
common: finish new isa support implementation

### PR DESCRIPTION
**Pull request by tomaszkapela**
VEX now supports and discerns PCOMMI/CLFLUSH/CLFLUSHOPT/CLWB and different
types of fences.

Unfortunately my IDE removed trailing whitespaces, hence the high number of changed lines.

### CODE REVIEW

File: VEX/pub/libvex_ir.h [hunk 1]
```diff
@@ -2410,7 +2410,11 @@ typedef
          back end, just as LL and SC themselves are. */
       Imbe_CancelReservation,
       /* Needed only on amd64.  It drains the iMC cache. */
-      Imbe_Drain
+      Imbe_Drain,
+      /* Might be interesting for some kinds of tools */
+      Imbe_SFence,
+      Imbe_MFence,
```
Comment (krzycz) [inline VEX/pub/libvex_ir.h]: Do we still need _Imbe_Fence_? I guess it should be equivalent (alias) to _Imbe_MFence_.

File: VEX/priv/guest_amd64_toIR.c [hunk 2]
```diff
@@ -13490,26 +13490,33 @@ Long dis_ESC_0F__SSE2 ( Bool* decode_OK,
          delta += 1;
          /* Insert a memory fence.  It's sometimes important that these
             are carried through to the generated code. */
-         stmt( IRStmt_MBE(Imbe_Fence) );
+         stmt( IRStmt_MBE(Imbe_SFence) );
```
Comment (krzycz) [inline VEX/priv/guest_amd64_toIR.c:13493]: What about adding an additional parameter to MBE statement - a fence kind? (Same as for Flush).

File: VEX/pub/libvex_ir.h (hunk 1 above)
Comment (tomaszkapela) [inline VEX/pub/libvex_ir.h]: Ok, I'll get rid of the MFence and tread Fence as MFence

File: VEX/priv/guest_amd64_toIR.c (hunk 2 above)
Comment (tomaszkapela) [inline VEX/priv/guest_amd64_toIR.c:13493]: This would cause quite a bit of rework, the same event type is reused in different architectures.

**Pull request closed without merging**